### PR TITLE
prevent totems dodge/parry

### PR DESCRIPTION
**Pull request by Gamemechanicwow**
## 🍰 Pullrequest

Prevent totems from parrying and dodging incoming attacks.

- GetUnitParryChance() was unexpectedly returning 5 for totems. This is changed to 0.
- RollMeleeOutcomeAgainst() no longer uses skilldifference for calculating dodge/parry for totems.

### Proof

- None

### Issues

- None

### How2Test

- Log onto 2 accounts. One of them has to be a shaman - the other has to be lower level than the shaman
- Shaman targets player on different account and types .m mainspeed 10
- Shaman places totem, targets it and types .m hp 99999
- Player on 2nd account autoattacks totem
- Notice if dodges/parries occurs

### CODE REVIEW

File: src/game/Objects/Unit.cpp [hunk 1]
```diff
@@ -2145,14 +2145,15 @@ MeleeHitOutcome Unit::RollMeleeOutcomeAgainst(Unit const* pVictim, WeaponAttackT
     }
 
     bool from_behind = !pVictim->HasInArc(this);
+    bool victimIsTotem = ((Creature const*)pVictim)->IsTotem();
```
Comment (0blu) [inline src/game/Objects/Unit.cpp:2148]: Is that a valid cast?

**Comment by ratkosrb** [comment]
Isn't this handled by CREATURE_STATIC_FLAG_NO_DEFENSE ?

**Comment by balakethelock** [comment]
CREATURE_STATIC_FLAG_NO_DEFENSE is currently implemented as making the unit have 0 defense skill, still able to dodge or parry albeit at a very low chance if the attacker has 300 weapon skill. This may or may not be accurate to vanilla.

**Comment by Gamemechanicwow** [comment]
> Isn't this handled by CREATURE_STATIC_FLAG_NO_DEFENSE ?

Not entirely.

https://github.com/vmangos/core/blob/development/src/game/Objects/Unit.cpp#L2254
dodge_chance  = 0 and dodgeSkillBonus can be negative, thereby resulting in a positive dodgechance

https://github.com/vmangos/core/blob/development/src/game/Objects/Unit.cpp#L2270
parry is not affected by this.

its a different issue imo. closing this one.

**Comment by 0blu** [comment]
@Gamemechanicwow Why was this issue closed? If the current solution doesn't address all cases and your PR offers a valid fix, we should consider merging it.

**Comment by ratkosrb** [comment]
I think the static flag should be made to prevent dodge/parry, it's likely what it's supposed to do.

**Comment by Gamemechanicwow** [comment]
> @Gamemechanicwow Why was this issue closed? If the current solution doesn't address all cases and your PR offers a valid fix, we should consider merging it.

i think it's preferred to figure out first if a unit with CREATURE_STATIC_FLAG_NO_DEFENSE should be able to gain additional avoidance based on leveldifference. If it shouldn't it would be a much cleaner fix imo and would span wider than just totems.

**Pull request closed without merging**